app/assets/src/components/common/DetailsSidebar/PipelineStepDetailsMode/PipelineStepDetailsMode.jsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import cs from "./pipeline_step_details_mode.scss";
+
+class PipelineStepDetailsMode extends React.Component {
+  renderInputFiles() {
+    const { inputFiles } = this.props;
+    if (inputFiles && inputFiles.length) {
+      const fromStepNameToFile = {};
+      inputFiles.forEach(file => {
+        if (!(file.fromStepName in fromStepNameToFile)) {
```

Comment:
you might want to use hasOwnProperty instead of `in` http://adripofjavascript.com/blog/drips/the-uses-of-in-vs-hasownproperty.html